en/tutorials/rag-blueprint.md

```diff
@@ -0,0 +1,1475 @@
+
+Example training output:
+
+```txt
+------------------------------------------------------------
+             Cross-Validation Results (5-Fold)             
+------------------------------------------------------------
+Metric             | Mean               | Std Dev           
+------------------------------------------------------------
+Accuracy           | 0.9214             | 0.0664            
+ROC AUC            | 0.9863             | 0.0197            
+------------------------------------------------------------
+Overall CV AUC: 0.9249 • ACC: 0.9216
+------------------------------------------------------------
+```
+
+### Feature importance analysis
+
+The trained model reveals which features are most important for ranking quality. For our sample application, the top features include:
+
+| Feature                     | Importance |
+| --------------------------- | ---------- |
+| nativeProximity             | 168.8498   |
+| firstPhase                  | 151.7382   |
+| max_chunk_sim_scores        | 69.4377    |
+| avg_top_3_chunk_text_scores | 56.5079    |
+| avg_top_3_chunk_sim_scores  | 31.8700    |
+| nativeRank                  | 20.0716    |
+| nativeFieldMatch            | 15.9914    |
+| elementSimilarity(chunks)   | 9.7003     |
+
+Key observations:
+
+* **Text proximity features** ([nativeProximity](../reference/nativerank.html#nativeProximity)) are highly valuable for understanding query-document relevance
+* **First-phase score** (`firstPhase`) being important validates that our first-phase ranking provides a good foundation
+* **Chunk-level features** (both text and semantic) contribute significantly to ranking quality
+* **Traditional text features** like [nativeRank](../reference/nativerank.html#nativeRank) and [bm25](../reference/bm25.html#ranking-function) remain important
+
+### Integrating the GBDT model into Vespa
+
+The trained LightGBM model is exported and added to your Vespa application package:
+
+```txt
+app/
+├── models/
+│   └── lightgbm_model.json
+```
+
+Create a new rank profile that uses this model:
+
+```txt
+rank-profile second-with-gbdt inherits collect-training-data {
+    ...
+
+    second-phase {
+        expression: lightgbm("lightgbm_model.json")
+    }
+
+    ...
+}
+```
+
+### Evaluating second-phase ranking performance
+
+Evaluate the GBDT-powered second-phase ranking on unseen test queries:
+
+```bash
+python evaluate_ranking.py --second_phase
+```
+
+Expected results show significant improvement over first-phase ranking:
+
+```json
+{
+    "accuracy@1": 0.9,
+    "accuracy@3": 0.95,
+    "accuracy@5": 1.0,
+    "accuracy@10": 1.0,
+    "precision@10": 0.23,
+    "recall@10": 0.93,
+    "precision@20": 0.13,
+    "recall@20": 0.99,
+    "mrr@10": 0.94,
+    "ndcg@10": 0.85,
+    "map@100": 0.78,
+    "searchtime_avg": 0.035
+}
+```
+
+This represents a dramatic improvement over first-phase ranking, with:
+
+* **accuracy@10** improving from 0.3 to 1.0
```

Comment:
I know I'm only supposed to write constructive comments here, but wow - these are nice improvements :)